ogc/display-ogc-api-collection/src/main/java/com/esri/samples/display_ogc_api_collection/DisplayOgcApiCollectionSample.java

```diff
@@ -0,0 +1,172 @@
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with the topographic basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
+
+      // create a map view and set the map to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create a progress indicator
+      var progressIndicator = new ProgressIndicator();
+
+      // add the map view and progress indicator to the stack pane
+      stackPane.getChildren().addAll(mapView, progressIndicator);
+
+      // define strings for the service URL and collection id
+      // note that the service defines the collection id which can be accessed via OgcFeatureCollectionInfo.getCollectionId().
+      String serviceUrl = "https://demo.ldproxy.net/daraa";
+      String collectionId = "TransportationGroundCrv";
+
+      // create an OGC feature collection table from the service url and collection id
+      ogcFeatureCollectionTable = new OgcFeatureCollectionTable(serviceUrl, collectionId);
+
+      // set the feature request mode to manual (only manual is currently supported).
+      // in this mode, the table must be manually populated - panning and zooming won't request features automatically
+      ogcFeatureCollectionTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+      // load the table
+      ogcFeatureCollectionTable.loadAsync();
+
+      // ensure the feature collection table has loaded successfully before creating a feature layer from it to display on the map
+      ogcFeatureCollectionTable.addDoneLoadingListener(() -> {
+        if (ogcFeatureCollectionTable.getLoadStatus() == LoadStatus.LOADED) {
+
+          // create a feature layer and set a renderer to it to visualize the OGC API features
+          var featureLayer = new FeatureLayer(ogcFeatureCollectionTable);
+          var simpleRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, ColorUtil.colorToArgb(Color.BLUE), 3));
+          featureLayer.setRenderer(simpleRenderer);
+
+          // add the layer to the map
+          map.getOperationalLayers().add(featureLayer);
+
+          // zoom to a small area within the dataset by default
+          Envelope datasetExtent = ogcFeatureCollectionTable.getExtent();
+          if (datasetExtent != null && !datasetExtent.isEmpty()) {
+            mapView.setViewpointGeometryAsync(
+              new Envelope(datasetExtent.getCenter(), datasetExtent.getWidth() / 3, datasetExtent.getHeight() / 3));
+          }
+
+        } else {
+          // show an alert dialog if there is a loading failure
+          new Alert(Alert.AlertType.ERROR, "Failed to load OGC Feature Collection Table");
+        }
+      });
+
+      // once the map view navigation has completed, query the OGC API feature table for
+      // additional features within the new visible extent.
+      mapView.addNavigationChangedListener(e -> {
```

Comment:
Might make more sense to use `addViewpointChangedListener` since by the comment it is changes to the extent you are interested in. I think navigation changed could be triggered by something that doesn't change the extent e.g. mouse down/mouse up.

Reply:
You could probably check `isNavigating` to avoid view point changes during pan... assuming the last viewpoint change happens after navigating goes false.

I get why you ended up with navigation changed over viewpoint but that reasoning won't be obvious to the sample user. On the other hand explaining about viewpoint changed events during pan might not be that great either. I think an inquisitive user might well wonder why you didn't use viewpoint changed as it seems the obvious thing to use. See if 'isNavigating` works otherwise looks like navigation changed will have to do.